src/picongpu/include/particles/particleToGrid/derivedAttributes/LarmorEnergy.hpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2013-2015 Axel Huebl, Rene Widera, Richard Pausch
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "particles/particleToGrid/derivedAttributes/LarmorEnergy.def"
+
+#include "simulation_defines.hpp"
+
+
+namespace picongpu
+{
+namespace particleToGrid
+{
+namespace derivedAttributes
+{
+
+    HDINLINE float1_64
+    LarmorEnergy::getUnit() const
+    {
+        return UNIT_ENERGY;
+    }
+
+    template< class T_Particle >
+    DINLINE float_X
+    LarmorEnergy::operator()( T_Particle& particle ) const
+    {
+        /* read existing attributes */
+        const float3_X mom = particle[momentum_];
+        const float3_X mom_mt1 = particle[momentumPrev1_];
+        const float_X weighting = particle[weighting_];
+        const float_X charge = attribute::getCharge( weighting, particle );
+        const float_X mass = attribute::getMass( weighting, particle );
+
+        /* calculate new attribute */
+        Gamma<float_X> calcGamma;
+        const typename Gamma<float_X>::valueType gamma = calcGamma( mom, mass );
+        const float_X c2 = SPEED_OF_LIGHT * SPEED_OF_LIGHT;
+
+        const float3_X mom_dt = mom - mom_mt1;
+        const float_X el_factor = charge * charge
```

Comment:
Just to add the result of the *offline discussion*:
Based on Jackson (14.26):
![001](https://cloud.githubusercontent.com/assets/5121158/11300566/76efc860-8f90-11e5-8f6e-4b99f1bc70cb.png)
which is in cgs-units, we get the following SI-unit equation:
![002](https://cloud.githubusercontent.com/assets/5121158/11300609/c5e1b672-8f90-11e5-8d4e-bbe92c307437.png)